File: Fibonachi a0a1a2a3a4a5/Without array/Java/fib.java
import java.util.Scanner;
public class fib{
	public static void main(String[] args) {
		int i, quantity, current, previous, sum;
		Scanner scan = new Scanner(System.in);
		i=1;
		quantity=scan.nextInt();
		current=1;
		previous=0;
		sum=0;
		String result_string = "" + current;
		while (i<quantity){
			sum=current;
			current=current+previous;
			previous=sum;
			i=i+1;
			result_string=result_string+current;
		}
		System.out.println(result_string);
	}
}

File: Fibonachi a0a1a2a3a4a5/Without array/Ruby/fib.rb
quantity=gets()
quantity=quantity.to_i
previous=0
current=1
sum=0
i=1
i=i.to_i
result_string=""+current.to_s
while (i<quantity)
	sum=current
	current=current+previous
	previous=sum
	i=i+1
	result_string=result_string+current.to_s
end
puts(result_string)

